File: selenium-project/src/main/java/com/master/seleniumproject/pages/login/LoginPage.java
package com.master.seleniumproject.pages.login;
import com.master.seleniumproject.pages.AbstractPage;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class LoginPage extends AbstractPage {
    @FindBy(css = "input[id='email']")
    private WebElement usernameInput;

    @FindBy(css = "input[id='password']")
    public WebElement passwordInput;

    @FindBy(css = "button[type='submit']")
    private WebElement loginButton;

    public LoginPage(WebDriver driver) {
        super(driver);
        PageFactory.initElements(driver, this);
    }

    public void fillUsernameInput(String username) {
        this.fillInput(this.usernameInput, username);
    }

    public void fillPasswordInput(String username) {
        this.fillInput(this.passwordInput, username);
    }

    public void clickLoginButton() {
        this.loginButton.click();
    }

}

File: playwright-project/src/pages/login/loginPage.ts
import { Locator, Page } from '@playwright/test';
import { BasePage } from '@pages/basePage';
import { ElementState } from '@constants/elementState';

export class LoginPage extends BasePage {
    constructor(
        page: Page,
        private usernameInput: Locator = page.locator('input[id=email]'),
        private passwordInput: Locator = page.locator('input[id=password]'),
        private loginButton: Locator = page.getByRole('button', { name: 'Login' }),
    ) {
        super(page);
    }

    async openLoginPage(): Promise<void> {
        await this.openURL('/');
    }

    async fillUsernameInput(username: string): Promise<void> {
        await this.usernameInput.fill(username);
    }

    async fillPasswordInput(password: string): Promise<void> {
        await this.passwordInput.fill(password);
    }

    async clickLoginButton(): Promise<void> {
        await this.loginButton.click();
    }

    async waitForLoginPageOpened(): Promise<void> {
        await this.usernameInput.waitFor({ state: ElementState.VISIBLE });
    }
}
